internal/providers/provider.go

```diff
@@ -385,6 +393,42 @@ type ImportedResource struct {
 	Private []byte
 }
 
+type MoveResourceStateRequest struct {
+	// SourceProviderAddress is the address of the provider that the resource
+	// is being moved from.
+	SourceProviderAddress string
+
+	// SourceTypeName is the name of the resource type that the resource is
+	// being moved from.
+	SourceTypeName string
+
+	// SourceSchemaVersion is the schema version of the resource type that the
+	// resource is being moved from.
+	SourceSchemaVersion int64
+
+	// SourceStateJSON and SourceStateFlatmap contain the state of the resource
+	// that is being moved. Because the schema is unknown, this contains only
+	// the raw data as stored in the state.
+	// SourceStateJSON is the current json state encoding.
+	// SourceStateFlatmap is the legacy flatmap encoding.
+	// Only one of these fields may be set for the move request.
+	SourceStateJSON    []byte
+	SourceStateFlatmap map[string]string
```

Comment:
I would drop this field, as there's no way a new provider should ever see (or be equipped to deal with) the pre-v0.12 encoding

Reply:
Done!